### ISSUE TITLE
A3CTrainer not passing information from batch to episode: untimeWarning: Mean of empty slice. out=out, **kwargs)

### ISSUE THREAD

**Issue by saltypeanuts** [ray-project/ray#5253]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10 Home 17134.885, WSL Debian
PRETTY_NAME="Debian GNU/Linux 9 (stretch)"
NAME="Debian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
ID=debian
- **Ray installed from (source or binary)**: source (pip3 install)
- **Ray version**: 0.7.2
- **Python version**: 3.5.3
- **Exact command to reproduce**:

### Describe the problem

The A3C trainer is correctly receiving reward information from batches but failing to correctly send it to the (episode?) results. I am using a custom environment.

From the output log, batch 10 (last batch) of first training cycle (0) where the reward is being calculated correctly:

` (pid=1283)             'prev_rewards': np.ndarray((10,), dtype=float32, min=0.0, max=3717.557, mean=3344.04),`

About halfway through the log:

```
/home/my_user/.local/lib/python3.5/site-packages/numpy/core/fromnumeric.py:3118: RuntimeWarning: Mean of empty slice.
  out=out, **kwargs)
/home/my_user/.local/lib/python3.5/site-packages/numpy/core/_methods.py:85: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
```

After this double_scalars issue there are a lot of nans passed through to the episode results.
`
{'sampler_perf': {}, 'timestamp': 1563897516, 'num_healthy_workers': 1, 'hostname': 'DESKTOP-53D6NM7', 'num_metric_batches_dropped': 0, 'phase': 1, 'episode_reward_min': nan, 'episode_len_mean': nan, 'off_policy_estimator': {}, 'time_this_iter_s': 40.374911308288574, 'episode_reward_mean': nan, 'timesteps_total': 2000, 'iterations_since_restore': 2, 'episode_reward_max': nan, 'done': False, 'experiment_id': 'e4d610eb4d2f4a6b966f47d3b6a9a996', 'time_total_s': 105.98298406600952, 'node_ip': '10.0.0.3', 'training_iteration': 2, 'episodes_this_iter': 0, 'custom_metrics': {}, 'timesteps_since_restore': 2000, 'episodes_total': 0, 'config': {'gamma': 0.99, 'env_config': {'df':                           low      high      open     close      volume  \
`

~~After the nans, the second iteration doesn't take place.~~ The trainer runs correctly using PPOTrainer in lieu of A3CTrainer.

Code calling the trainer:
```
    trainer = A3CTrainer(env = "my_env", config = config)
    for _ in range(2):
        print(_)
        result = trainer.train()
        result["phase"] = 1
        print(result)
```
### Source code / logs
Full logs from trial (with dataframe omitted)
https://pastebin.com/WKU8Dnz1

**Comment by ericl** [comment]
The 'nan' just means no episodes have completed yet. If you keep training it for more iterations some stats should show up?

**Comment by saltypeanuts** [comment]
> 
> 
> The 'nan' just means no episodes have completed yet. If you keep training it for more iterations some stats should show up?

It's gone through two train() cycles by the end of the code block which I would assume each "train()" command would be an episode. I guess I'm confused on how you're reporting 'episodes' as it's not intuitive that "train()" is not considered an episode.

**Comment by ericl** [comment]
Nope, train is just a distributed training step, it can be many episodes,
or just a fraction of a number of episodes. In distributed execution, there
are many concurrent episodes and there aren't any guarantees you can
provide on whether a particular number have finished or not.

Eric
